app/com/lucidchart/open/cashy/controllers/UploadController.scala

```diff
@@ -212,17 +195,17 @@ class UploadController extends AppController with ExtensionsConfig with UploadFe
     Action(parse.multipartFormData(FileHandler.handleFilePartAsByteArray)) { implicit request =>
       val response = uploadForm.bindFromRequest.fold(
         formWithErrors => formWithErrors.errors,
-        data => List.empty
+        data => {
+          Try {
+            AssetDataHelper.getData(request.body.file("assetFile"), data.assetURL)
```

Comment:
It's 3x if you're doing an image resize (kraken preview, validate, then upload).  I'll see how feasible it is to reduce that.